planar/triangulate/delaunaytriangulationbuilder.go

```diff
@@ -0,0 +1,214 @@
+ *
+ * @param coords a List of Coordinates
+ * @return the envelope of the set of coordinates
+public static Envelope envelope(Collection coords)
+{
+	Envelope env = new Envelope();
+	for (Iterator i = coords.iterator(); i.hasNext(); ) {
+		Coordinate coord = (Coordinate) i.next();
+		env.expandToInclude(coord);
+	}
+	return env;
+}
+*/
+
+/*
+Sets the sites (vertices) which will be triangulated.
+All vertices of the given geometry will be used as sites.
+
+geom - the geometry from which the sites will be extracted.
+*/
+func (dtb *DelaunayTriangulationBuilder) SetSites(g geom.Geometry) error {
+	// remove any duplicate points (they will cause the triangulation to fail)
+	c, err := dtb.extractUniqueCoordinates(g)
+	dtb.siteCoords = c
+	return err
+}
+
+/**
+ * Sets the sites (vertices) which will be triangulated
+ * from a collection of {@link Coordinate}s.
+ *
+ * @param coords a collection of Coordinates.
+public void setSites(Collection coords)
+{
+	// remove any duplicate points (they will cause the triangulation to fail)
+	siteCoords = unique(CoordinateArrays.toCoordinateArray(coords));
+}
+*/
+
+/**
+ * Sets the snapping tolerance which will be used
+ * to improved the robustness of the triangulation computation.
+ * A tolerance of 0.0 specifies that no snapping will take place.
+ *
+ * @param tolerance the tolerance distance to use
+public void setTolerance(double tolerance)
+{
+	this.tolerance = tolerance;
+}
+*/
+
+/*
+create will create the triangulation.
+
+return true on success, false on failure.
+*/
+func (dtb *DelaunayTriangulationBuilder) create() bool {
+	if dtb.subdiv != nil {
+		return true
+	}
+	if len(dtb.siteCoords) == 0 {
+		return false
+	}
+
+	var siteEnv *geom.Extent
+	for _, v := range dtb.siteCoords {
+		if siteEnv == nil {
+			siteEnv = geom.NewExtent(v)
+		}
+		siteEnv.AddGeometry(v)
+	}
+
+	dtb.subdiv = quadedge.NewQuadEdgeSubdivision(*siteEnv, dtb.tolerance)
+	triangulator := new(IncrementalDelaunayTriangulator)
+	triangulator.subdiv = dtb.subdiv
+	triangulator.InsertSites(dtb.siteCoords)
+
+	return true
+}
+
+/**
+ * Gets the {@link QuadEdgeSubdivision} which models the computed triangulation.
+ *
+ * @return the subdivision containing the triangulation
+public QuadEdgeSubdivision getSubdivision()
+{
+	create();
+	return subdiv;
+}
+*/
+
+/*
+GetEdges gets the edges of the computed triangulation as a MultiLineString.
+
+returns the edges of the triangulation
+*/
+func (dtb *DelaunayTriangulationBuilder) getEdges() geom.MultiLineString {
+	if !dtb.create() {
+		return geom.MultiLineString{}
+	}
+	return dtb.subdiv.GetEdgesAsMultiLineString()
+}
+
+/*
+GetTriangles Gets the faces of the computed triangulation as a
+GeometryCollection Polygons.
+
+Unlike JTS, this method returns a MultiPolygon. I found not all viewers like
+displaying collections. -JRS
+*/
+func (dtb *DelaunayTriangulationBuilder) GetTriangles() (geom.MultiPolygon, error) {
+	if !dtb.create() {
+		return geom.MultiPolygon{}, nil
```

Comment:
We should add  a check for `dtb == nil` or document that call `GetTriangles` on a nil `dtb` will panic.